Websites/perf.webkit.org/public/api/build-requests.php

```diff
@@ -76,8 +76,7 @@ function update_builds($db, $updates) {
             $is_build = $request_row['request_order'] < 0;
             if ($is_build) {
                 $db->query_and_fetch_all('UPDATE build_requests SET request_status = \'failed\'
-                    WHERE request_group = $1 AND request_order > $2',
-                    array($request_row['request_group'], $request_row['request_order']));
+                    WHERE request_group = $1 AND request_order >= 0', array($request_row['request_group']));
```

Comment:
This doesn't seem right? If one of the build steps had failed, when we need to mark all other building requests as failing. Otherwise, we'd leave incomplete build requests.

Reply:
Discussed in person that we will keep this change so that even one of the build type requests fails (could be build failure or missing entry from buildbot), the other build type requests can still be scheduled / updated so that build products from those requests can be potentially reused or expose build configuration issues (e.g. compilation error) earlier.
To avoid regressing the time users receive build failure notifications, we will change `/api/test-groups/ready-for-notification` to qualify a test group's "ready-for-notification" if all test type requests are in a final state (completed / failed).